packages/primitives/src/Link/Link.tsx

```diff
@@ -0,0 +1,68 @@
+import styled from '@emotion/styled'
+import { IntrinsicSxElements } from 'theme-ui'
+import { createShouldForwardProp } from '@styled-system/should-forward-prop'
+import {
+  space,
+  SpaceProps,
+  color,
+  ColorProps,
+  typography,
+  TypographyProps,
+  layout,
+  LayoutProps,
+  flexbox,
+  background,
+  BackgroundProps,
+  border,
+  BorderProps,
+  shadow,
+  ShadowProps,
+} from 'styled-system'
+import { variant, sx } from '../Box'
+/**
+ * propNames are typed as string[] | undefined. Undefined is not
+ * an iterator so we have to cast propNames to only a string[]
+ */
+const shouldForwardProp = createShouldForwardProp([
+  ...(space.propNames as string[]),
+  ...(color.propNames as string[]),
+  ...(typography.propNames as string[]),
+  ...(layout.propNames as string[]),
```

Comment:
Do we need layout? I'm not sure how often devs are going to be changing the width, height, and display values of links.